components/formats-bsd/src/loci/formats/tiff/TiffParser.java

```diff
@@ -229,14 +229,6 @@ public IFDList getIFDs() throws IOException {
         subOffsets = ifd.getIFDLongArray(IFD.SUB_IFD);
       }
       catch (FormatException e) { }
-      if (subOffsets != null) {
```

Comment:
Given that it's a special-case hack used solely by the Canon DNG and Nikon TIFF readers, it's exceedingly unlikely any existing reader will be broken.  It would require input images with SubIFDs present, and a requirement on the part of the reader to expect those SubIFDs to be placed at specific offsets in the IFD list.  The chance of this esoteric combination is almost nearly zero.  We've fixed the instances of its usage in our own code in this PR, and the chance of breakage for external readers is so small as to be almost nonexistent.

Reply:
I think we all agree on using this occasion to improve the behavior of this API to make it clear and unambiguous. As mentioned yesterday, I am fine with merging of this PR to make progress on the OME-TIFF pyramid reading/writing and exploring the different options when this work gets reintegrated into the mainline.